### DISCUSSION TITLE
How to use a plugin (microphone) with npm?

### DISCUSSION THREAD

**Discussion by Tobjoern** [katspaugh/wavesurfer.js#2765] [Q&A] [answered]
Hello,
I'm trying to create a voice-recorder with wavesurfer for my next.js app.
I already installed wavesurfer, and I'm able to display normal audio recordings.
But I'm unable to get the microphone plugin to work (https://wavesurfer-js.org/example/microphone/index.html).

The following code throws an error:

```
import Wavesurfer from 'wavesurfer.js';
Wavesurfer.microphone.create()
```

I'm looking forward to suggestions!

**Comment by katspaugh** [comment, marked as answer]
I'd recommend trying out the [alpha version](https://github.com/katspaugh/wavesurfer.ts) which has a more full-featured [Record plugin](https://wavesurfer-ts.pages.dev/tutorial/#/examples/record.js).

**Reply by Tobjoern** [reply]
It works well, thanks!
